setup.py

```diff
@@ -397,6 +397,7 @@ def run(self):
               "html5lib",
               "junitxml",
               "tenacity",
+              "jaraco.functools",
```

Comment:
The dependency right above this new one, `tenacity`, is a library for doing retries.  Perhaps we can use that one.

That said, I think both of these libraries only work for synchronous tests.  If a test returns a Deferred and that Deferred fires with a failure, do they do anything at all?

Reply:
Good point. I wasn't aware these test functions might return Deferred results. You're right - if a Deferred is returned, it will get no retry behavior.